### PR TITLE
Feat/parse packet unit test

### PR DESCRIPTION
**Pull request by Tlabting**
Added additional unit tests for edl.py
I also included a filterwarning to ignore a deprecation warning on generating a bad packet, which is used in a unit test to trigger an invalid FECF exception.

### CODE REVIEW

File: tests/test_edl.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import unittest
-
-from oresat_c3.protocols.edl import EdlServer, EdlClient
-
+from unittest.mock import patch
+from oresat_c3.protocols.edl import EdlServer, EdlClient, EdlError, crc16_bytes
+from spacepackets.uslp.frame import TransferFrame, FrameType
```
[the remark this answers is not in the document]
Comment (ryanpdx) [reply tests/test_edl.py:4]: The `unittest` module is already imported, no reason to import `patch` directly

**Review by ryanpdx — CHANGES_REQUESTED** [review]
Looks great!

Just some minor changes and one question

File: tests/test_edl.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmitri-mcguckin) [reply tests/test_edl.py:4]: While I agree imports should occur at the top, importing directly or aliasing via the `as` keyword is useful if you don't want to have to type `unittest.mock.patch.patch(...)` every time. Though if the reference is too vague and you'd like to know in the call which module `patch()` refers to. Maybe `import unittest.mock as um` is a good compromise?

Just food for thought.

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,8 @@
 [build-system]
 requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
+[pytest.ini_options]
+filterwarnings = [
+    #Specifically creating a bad packet for unit test, ignore pytest warning on this specific test
+    "ignore::DeprecationWarning:test_edl.TestEdl.test_parse_packet_invalid_fecf"
+]
```
[the remark this answers is not in the document]
Comment (Tlabting) [reply pyproject.toml:8]: Yes, I was aware. I was getting a warning for a bad packet in the unit test where I purposefully create a bad packet. I was under the impression that the code I wrote was disabling the warning generated by pytest for the bad packet in response to the one created for the purpose of testing. Please correct me if I'm mistaken, I appreciate it!

File: tests/test_edl.py [hunk 3]
```diff
@@ -21,3 +21,113 @@ def test_telecommand(self):
...
+       req_packet = client.generate_request(payload)
+
+       #Modifying the packet length in the primary header so that it's invalid
+       invalid_packet_len = 255
+       invalid_packet = req_packet[:4] + invalid_packet_len.to_bytes(2, 'big') + req_packet[6:]
+       #mock_crc16_bytes.return_value = invalid_packet[-2:]
+
+       #Checking if EdlError is raised for invalid packet or fram length
+       #Use patch as a context manager
+       with patch("oresat_c3.protocols.edl.crc16_bytes") as mock_crc16_bytes:
```
[the remark this answers is not in the document]
Comment (Tlabting) [reply tests/test_edl.py:80]: The reason I decided to use patch here is to mock out the return value for the crc16_bytes function call in edl.py on line 269 in order to test for the UslpInvalidRawPacketOrFrameLen Exception. If I did not mock this, the Exception on line 271 was caught instead. If there is a better way to do it, please let me know. I am always eager to learn new techniques!

**Review by ryanpdx — APPROVED** [review]

**Pull request merged**